File: src/main/java/xyz/oribuin/eternalclaims/manager/CommandManager.java
package xyz.oribuin.eternalclaims.manager;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.command.framework.RoseCommandWrapper;
import dev.rosewood.rosegarden.manager.AbstractCommandManager;
import xyz.oribuin.eternalclaims.command.ClaimsCommandWrapper;

import java.util.List;

public class CommandManager extends AbstractCommandManager {

    public CommandManager(RosePlugin rosePlugin) {
        super(rosePlugin);
    }

    @Override
    public List<Class<? extends RoseCommandWrapper>> getRootCommands() {
        return List.of(ClaimsCommandWrapper.class);
    }

    @Override
    public List<String> getArgumentHandlerPackages() {
        return List.of("xyz.oribuin.eternalclaims.command.argument");
    }

}

File: src/main/kotlin/dev/oribuin/eternalclaims/manager/CommandManager.kt
package dev.oribuin.eternalclaims.manager

import dev.oribuin.eternalclaims.command.BaseCommand
import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.command.framework.BaseRoseCommand
import dev.rosewood.rosegarden.manager.AbstractCommandManager
import java.util.function.Function

class CommandManager(rosePlugin: RosePlugin) : AbstractCommandManager(rosePlugin) {

    override fun getRootCommands(): List<Function<RosePlugin, BaseRoseCommand>> {
        return listOf(Function { BaseCommand(it) })
    }

}
